dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/DefaultTrackerProgramRuleService.java

```diff
@@ -79,7 +82,12 @@ public DefaultTrackerProgramRuleService(
             .collect( Collectors.toMap( Enrollment::getEnrollment, e -> {
                 ProgramInstance enrollment = enrollmentTrackerConverterService.fromForRuleEngine( bundle.getPreheat(),
                     e );
-                return programRuleEngine.evaluate( enrollment, Sets.newHashSet() );
+                try{
+                    return programRuleEngine.evaluate( enrollment, Sets.newHashSet() );
+                } catch (Exception ex) {
```

Comment:
I'm not sure about the message: I would not reference the payload, since we are catching a generic exception. Perhaps something along the line: "An error occured during an Program RUle engine call for enrollment ..... Please check the response payload for additional information"